src/Planners/src/CubicSpline.cpp

```diff
@@ -728,14 +748,15 @@ bool CubicSpline::evaluatePoint(const double& t,
 
     if (t >= m_pimpl->polynomials.back().finalPoint->timeInstant)
     {
-        const double finalTime = m_pimpl->polynomials.back().finalPoint->timeInstant
-                                 - m_pimpl->polynomials.back().initialPoint->timeInstant;
+        const auto finalTime = m_pimpl->polynomials.back().finalPoint->timeInstant
+                               - m_pimpl->polynomials.back().initialPoint->timeInstant;
         m_pimpl->getPositionAtTime(finalTime, m_pimpl->polynomials.back(), position);
         m_pimpl->getVelocityAtTime(finalTime, m_pimpl->polynomials.back(), velocity);
         m_pimpl->getAccelerationAtTime(finalTime, m_pimpl->polynomials.back(), acceleration);
         return true;
     }
 
+    // TODO(Giulio) Change this algorithm with the reverse iterator
```

Comment:
Note that in the end, its complexity is always linear in the dimension of the vector.

Reply:
That's true